src/Http/Http.Abstractions/src/Metadata/IProducesResponseTypeMetadata.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+namespace Microsoft.AspNetCore.Http.Metadata
+{
+    /// <summary>
+    /// Defines a contract for outline the response type returned from an endpoint.
+    /// </summary>
+    public interface IProducesResponseTypeMetadata
+    {
+        /// <summary>
+        /// Gets the optimistic return type of the action.
+        /// </summary>
+        Type? Type { get; }
+
+        /// <summary>
+        /// Gets the HTTP status code of the response.
+        /// </summary>
+        int StatusCode { get; }
+
+        /// <summary>
+        /// Configures a collection of allowed content types which can be produced.
+        /// </summary>
+        void SetContentTypes(ICollection<string> contentTypes);
```

Comment:
After spending time looking at this, I'm not sure why this isn't a property:

```suggestion
       IReadOnlyCollection<string> ContentTypes { get; }
```

@pranavkm ?